File: sa31/2024-11-13/project/JavaLesson2/src/GirlFriend.mjs
/**
 * SA31 課題2
 * サポートJavaScriptクラス
 *
 * ガールフレンドを表すクラス
 *
 * @author Shinzo SAITO
 */
export default class GirlFriend {
    /**
     * @type {string} 名前を表すフィールド
     */
    #name;
    /**
     * @type {number} 誕生年を表すフィールド
     */
    #birthYear;
    /**
     * @type {number} 誕生月を表すフィールド
     */
    #birthMonth;
    /**
     * @type {number} 誕生日を表すフィールド
     */
    #birthDay;

    /**
     * コンストラクタ
     *
     * @param {string} name 名前
     * @param {number} birthYear 誕生年
     * @param {number} birthMonth 誕生月
     * @param {number} birthDay 誕生日
     */
    constructor(name, birthYear, birthMonth, birthDay) {
        this.#name = name;
        this.#birthYear = birthYear;
        this.#birthMonth = birthMonth;
        this.#birthDay = birthDay;
    }

    /**
     * 引数で渡された月が誕生月かどうかを判定するメソッド
     *
     * @param {number} month 判定に使われる月
     * @returns {boolean} 誕生月の場合はtrue、そうでない場合はfalse
     */
    isBirthMonth(month) {
        if (month === this.#birthMonth) {
            return true;
        } else {
            return false;
        }
    }

    /**
     * 引数で渡された年に何歳になるか計算するメソッド
     *
     * @param {number} year 計算の基となる年
     * @returns {number} 年齢
     */
    getAge(year) {
        return year - this.#birthYear;
    }

    /**
     * 名前フィールドのゲッタ
     * @returns {string} 名前
     */
    get name() {
        return this.#name;
    }

    /**
     * 誕生年フィールドのゲッタ
     * @returns {number} 誕生年
     */
    get birthYear() {
        return this.#birthYear;
    }

    /**
     * 誕生月フィールドのゲッタ
     * @returns {number} 誕生月
     */
    get birthMonth() {
        return this.#birthMonth;
    }

    /**
     * 誕生日フィールドのゲッタ
     * @returns {number} 誕生日
     */
    get birthDay() {
        return this.#birthDay;
    }
}

File: sa31/2024-11-13/project/JavaLesson2/src/local/hal/sa31/androidjava/lesson1/asgmt/GirlFriend.java
package local.hal.sa31.androidjava.lesson1.asgmt;

/**
 * SA31 課題2
 * サポートJavaクラス
 *
 * ガールフレンドを表すクラス
 *
 * @author Kosuke Jimbo
 */
public class GirlFriend {
    private final String _name;
    private final int _birthYear;
    private final int _birthMonth;
    private final int _birthDay;

    /**
     * コンストラクタ
     *
     * @param name 名前
     * @param birthYear 誕生年
     * @param birthMonth 誕生月
     * @param birthDay 誕生日
     */
    GirlFriend(String name, int birthYear, int birthMonth, int birthDay) {
        _name = name;
        _birthYear = birthYear;
        _birthMonth = birthMonth;
        _birthDay = birthDay;
    }

    /**
     * 引数で渡された月が誕生月かどうかを判定するメソッド
     *
     * @param month 判定に使われる月
     * @return 誕生月の場合はtrue、そうでない場合はfalse
     */
    public Boolean isBirthMonth(int month) {
        if (month == _birthMonth) {
            return true;
        } else {
            return false;
        }
    }

    /**
     * 引数で渡された年に何歳になるか計算するメソッド
     *
     * @param year 計算の基となる年
     * @return 年齢
     */
    public int getAge(int year) {
        return year - _birthYear;
    }

    /**
     * 名前フィールドのゲッタ
     */
    public String getName() {
        return _name;
    }

    /**
     * 誕生年フィールドのゲッタ
     */
    public int getBirthYear() {
        return _birthYear;
    }

    /**
     * 誕生月フィールドのゲッタ
     */
    public int getBirthMonth() {
        return _birthMonth;
    }

    /**
     * 誕生日フィールドのゲッタ
     */
    public int getBirthDay() {
        return _birthDay;
    }
}
